### PR TITLE
Bug 1496356 - Outdated geckoview-nightly-x86/maven-metadata.xml

### PR DESCRIPTION
**Pull request by JohanLorenzo**
I tested this patch manually, thanks the Web IDE integrated in AWS Lambda. I plan to write unit tests to get the script tested.

### CODE REVIEW

File: lambda.py [hunk 1]
```diff
@@ -87,8 +110,7 @@ def generate_release_maven_metadata(folder_content_keys):
 
 def generate_versions(folder_content_keys):
     return sorted(list(set([
-        get_version(key) for key in folder_content_keys
-        if key
+        get_version(key) for key in folder_content_keys if key
```
Comment (MihaiTabara) [inline lambda.py]: 👍 for fixing formatting!

File: lambda.py [hunk 2]
```diff
@@ -18,38 +20,59 @@
...
-        file.key for file in bucket.objects.all()
-        if file.key.startswith(folder_key) and '/' in file.key.strip('{}/'.format(folder_key))
+        file.key for file in bucket.objects.filter(Prefix=folder_key)
+        if file.key.endswith('.pom')
     ]
 
 
 def get_group_id(key):
-    return '.'.join(key.split('/')[:-3])
+    folder_names_but_last_three = key.split('/')[:-3]
```
Comment (MihaiTabara) [inline lambda.py]: Nit (not blocking, just an optional opinion) I'm not a huge fan of splitting lines with trailing `\` so I'm usually leaning for less python solutions in cases like this. Rewriting to following seems easier to read but, as said, not blocking and a totally personal preference:

```
if folder_names_but_last_three[0] in ('maven', 'maven2'):
   # an explanatory comment maybe?
   folder_names_without_maven_prefix = folder_names_but_last_three[1:]
else
   # an explanatory comment maybe?
   folder_names_without_maven_prefix = folder_names_but_last_three
```

File: lambda.py [hunk 3]
```diff
@@ -18,38 +20,59 @@
...
+    return folder_with_trailing_slash
 
 
-def list_content_in_subfolders(bucket, folder_key):
+def list_pom_files_in_subfolders(bucket, folder_key):
     return [
-        file.key for file in bucket.objects.all()
-        if file.key.startswith(folder_key) and '/' in file.key.strip('{}/'.format(folder_key))
+        file.key for file in bucket.objects.filter(Prefix=folder_key)
+        if file.key.endswith('.pom')
```
Comment (MihaiTabara) [inline lambda.py]: I better like this filtering, nice! 👍

File: lambda.py (hunk 2 above)
Comment (JohanLorenzo) [reply lambda.py]: Done!

Comment (MihaiTabara) [reply lambda.py]: Sweet!

**Pull request merged**